pkg/controller/hibernation/hibernation_controller_test.go

```diff
@@ -73,23 +81,51 @@ func TestReconcile(t *testing.T) {
 		{
 			name: "cluster not installed",
 			cd:   cdBuilder.Options(o.notInstalled, o.shouldHibernate).Build(),
+			cs:   csBuilder.Build(),
 			validate: func(t *testing.T, cd *hivev1.ClusterDeployment) {
 				require.Nil(t, getHibernatingCondition(cd))
 			},
 		},
 		{
 			name: "start hibernating, older version",
 			cd:   cdBuilder.Options(o.shouldHibernate, testcd.WithClusterVersion("4.3.11")).Build(),
+			cs:   csBuilder.Build(),
 			validate: func(t *testing.T, cd *hivev1.ClusterDeployment) {
 				cond := getHibernatingCondition(cd)
 				require.NotNil(t, cond)
 				assert.Equal(t, corev1.ConditionFalse, cond.Status)
 				assert.Equal(t, hivev1.UnsupportedHibernationReason, cond.Reason)
 			},
 		},
+		{
+			name: "start hibernating, syncsets not applied",
+			cd:   cdBuilder.Options(o.shouldHibernate, testcd.InstalledTimestamp(time.Now())).Build(),
+			cs:   csBuilder.Options(testcs.WithNoFirstSuccessTime()).Build(),
+			validate: func(t *testing.T, cd *hivev1.ClusterDeployment) {
+				cond := getHibernatingCondition(cd)
+				require.NotNil(t, cond)
+				assert.Equal(t, corev1.ConditionFalse, cond.Status)
+				assert.Equal(t, hivev1.SyncSetsNotAppliedReason, cond.Reason)
+			},
+		},
+		{
+			name: "start hibernating, syncsets not applied but 10 minutes have passed since cd install",
+			cd:   cdBuilder.Options(o.shouldHibernate, testcd.InstalledTimestamp(time.Now().Add(-15*time.Minute))).Build(),
+			cs:   csBuilder.Options(testcs.WithNoFirstSuccessTime()).Build(),
+			setupActuator: func(actuator *mock.MockHibernationActuator) {
+				actuator.EXPECT().StopMachines(gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(nil)
+			},
+			validate: func(t *testing.T, cd *hivev1.ClusterDeployment) {
+				cond := getHibernatingCondition(cd)
+				require.NotNil(t, cond)
+				assert.Equal(t, corev1.ConditionTrue, cond.Status)
+				assert.Equal(t, hivev1.StoppingHibernationReason, cond.Reason)
+			},
+		},
```

Comment:
Would it be feasible to add a test for when there are no syncsets to apply or is that covered elsewhere. I guess it's every other test isn't it.

Reply:
Also are we missing a test for when SyncSets have been applied successfully?

Reply:
Added a test to ensure `cs.Status.FirstSuccessTime` is set w/ no syncsets and another in hibernation for syncsets applied successfully.